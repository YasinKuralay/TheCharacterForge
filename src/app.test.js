import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import App from "./app.js";

it("renders the App component without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it("matches the previous snapshot", () => {
    const AppComponentSnapshot = renderer.create(<App />).toJSON();
    expect(AppComponentSnapshot).toMatchSnapshot();
});
