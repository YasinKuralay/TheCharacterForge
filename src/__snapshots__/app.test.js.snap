// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the previous snapshot of the App Component 1`] = `
Array [
  <div>
    <div
      id="topBarPlaceholder"
    />
    <div
      id="topBar"
    >
      <img
        id="topbarLogo"
        src="./logo.png"
        width="160px"
      />
      <div
        className="topBarComponent"
        onClick={[Function]}
      >
        <i
          className="fas fa-home fa-lg"
        />
        <span>
           Home
        </span>
      </div>
      <a
        href="/logout"
      >
        <div
          className="topBarComponent"
          id="logout"
        >
          <i
            className="fas fa-sign-out-alt fa-lg"
          />
          <span>
             Logout
          </span>
        </div>
      </a>
    </div>
  </div>,
  <div>
    <a
      href="/browseCharacters"
      onClick={[Function]}
    >
      <div
        id="browseCharacters"
      >
        <h2>
          Browse Your Characters
        </h2>
        <i
          className="fas fa-search fa-3x"
        />
      </div>
    </a>
    <a
      href="/createcharacter"
      onClick={[Function]}
    >
      <div
        id="createCharacters"
      >
        <h2>
          Create New Character
        </h2>
        <i
          className="fas fa-hat-wizard fa-3x"
        />
      </div>
    </a>
  </div>,
]
`;
